.github/workflows/e2e_test_ci.yml

```diff
@@ -352,4 +352,127 @@ jobs:
           if [[ $sql_output == *"error"* ]] || [[ $sql_output == *"not found"* ]]; then
             echo "Unexpected response from spice sql, failed to perform test query: $sql_output"
             exit 1
-          fi
\ No newline at end of file
+          fi
+  test_local_acceleration:
+    name: "E2E Test: acceleration on ${{ matrix.name }} using ${{ matrix.acceleration.engine }}"
+    runs-on: ${{ matrix.runner }}
+
+
+    strategy:
+      fail-fast: false
+      matrix:
+        acceleration: [
+          { engine: arrow, mode: memory},
+          { engine: duckdb, mode: memory},
+          { engine: duckdb, mode: file},
+          { engine: sqlite, mode: memory},
+          { engine: sqlite, mode: file},
+          { engine: postgres},
+        ]
+        include:
+          - name: Linux x64
+            runner: ubuntu-latest
+            target_os: linux
+            target_arch: x86_64
+            target_arch_go: amd64
+          - name: macOS aarch64 (Apple Silicon)
+            runner: macos-14
+            target_os: darwin
+            target_arch: aarch64
+            target_arch_go: arm64
+          - name: macOS x64 (Intel)
+            runner: macos-12
+            target_os: darwin
+            target_arch: x86_64
+            target_arch_go: amd64
+          # - name: Windows x64
+          #   runner: rust-windows-x64
+          #   target_os: windows
+          #   target_arch: x86_64
+          #   target_arch_go: amd64
+
+    steps:
+      - name: download artifacts - build_${{ matrix.target_os }}_${{ matrix.target_arch }}
+        uses: actions/download-artifact@v4
+        with:
+          name: build_${{ matrix.target_os }}_${{ matrix.target_arch }}
+          path: ./build
+
+      - name: Install spice
+        run: |
+          chmod +x ./build/spice
+          chmod +x ./build/spiced
+          echo "$GITHUB_WORKSPACE/build" >> $GITHUB_PATH
+
+      - name: Check spice version
+        run: spice version
+
+      - name: Spice dataset configure
+        working-directory: test_app
+        run: |
+          ENGINE=$(echo '${{ matrix.acceleration.engine }}')
+          MODE=$(echo '${{ matrix.acceleration.mode }}')
+
+          
+          echo "datasets:" >> spicepod.yaml
```

Comment:
Can we pull the spicepods we use for e2e tests from a folder in the repo? The benefit to that is we have an easy place to go to for example spicepods. i.e. `cat $REPO_ROOT/test/spicepods/acceleration_arrow_memory.yaml`

Reply:
Like it 👍 - I'm going merge this and create separate PR for this approach-  will see how to make this approach re-usable so we can rely on it for other scenarios as well (not just acceleration), there might be some discussion around additional meta, for example target dataset name to verify (which is part of spicepod definition and can vary)